service/topology_coordinator.cc

```diff
@@ -2287,6 +2323,13 @@ future<bool> topology_coordinator::maybe_start_tablet_migration(group0_guard gua
 
     co_await generate_migration_updates(updates, guard, plan);
 
+    // Updates can be empty if only decision to finalize split was emitted, as it doesn't immediately
+    // cause changes to topology metadata. Updates only happen once handling of split finalize takes
+    // over the state machine.
+    if (updates.empty()) {
```

Comment:
Seems more indirect than necessary. Why not check `plan.resize_plan().finalize_resize` ?

Reply:
it can happen that plan maker decides to emit a migration for table A and also consider table B to be split ready. In this case, we don't want to transition state machine into tablet_split as there's migration work to do. so we're waiting for state machine to quiesce first.

Reply:
Good point.